packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -528,12 +499,108 @@ contract ControllerHelper is UniswapControllerHelper, AaveControllerHelper, IERC
 
             IController(controller).withdrawUniPositionToken(data.vaultId);
 
-            _closeUniLp(data);
+            (uint256 wPowerPerpAmountInLp, ) = ControllerHelperUtil.closeUniLp(
+                nonfungiblePositionManager,
+                ControllerHelperDataType.closeUniLpParams({
+                    tokenId: data.tokenId,
+                    liquidity: data.liquidity,
+                    liquidityPercentage: data.liquidityPercentage,
+                    amount0Min: data.amount0Min,
+                    amount1Min: data.amount1Min
+                }),
+                isWethToken0
+            );
 
-            _checkPartialLpClose(data.vaultId, data.tokenId, data.liquidityPercentage);
+            // close short position using amounts collected from closing LP, withdraw collateralToWithdraw + deposited collateralToFlashloan
+            _closeShortWithAmountsFromLp(
+                data.vaultId,
+                wPowerPerpAmountInLp,
+                data.wPowerPerpAmountToBurn,
+                data.collateralToWithdraw.add(data.collateralToFlashloan),
+                data.limitPriceEthPerPowerPerp
+            );
+            ControllerHelperUtil.checkPartialLpClose(nonfungiblePositionManager, controller, data.vaultId, data.tokenId, data.liquidityPercentage);
+        } 
+        else if (CALLBACK_SOURCE(_callSource) == CALLBACK_SOURCE.FLASHLOAN_REBALANCE_VAULT_NFT) {
```

Comment:
assume the rebalance where the ranges change need to be done here, or will be in a different function

Reply:
Here